boot/freeldr/freeldr/arch/i386/machxbox.c

```diff
@@ -17,12 +17,156 @@
  */
 
 #include <freeldr.h>
+#include <cportlib/cportlib.h>
 
 #include <debug.h>
 
 DBG_DEFAULT_CHANNEL(HWDETECT);
 
 
+extern
+VOID
+DetectSerialPointerPeripheral(PCONFIGURATION_COMPONENT_DATA ControllerKey, PUCHAR Base);
+
+static
+ULONG
+XboxGetSerialPort(UCHAR Device, PULONG Irq)
+{
```

Comment:
Is this different from the detection on PCs? If not, can we place the detection code in a static lib and share it between freeldr and kdcom?

Reply:
Yes, I also thought about it.

Reply:
Ok this is refactored now, the common part is in the single place, and hardware specific differences are placed separately.

Reply:
The kdcom for XBOX can be thought about and done in a separate PR :)